### PR TITLE
kafka(ticdc): index-value dispatcher support set index names.

### LINKED ISSUE #9821: index-value dispatcher allow user specify the index name
For Kafka sink changefeed, the `index-value` dispatcher calculates the partition number by hashing the handle key columns' value, which is implicit, and hard to explain.

Some user hopes that they can specify the index to calculate the partition number explicitly.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9821

### What is changed and how it works?

1. when create the changefeed, verify the dispatcher rule, if using `index-value` dispatcher and the `indexName` is set, make sure the corresponding index exists if there are any table at the upstream TiDB.
2. index-value dispatcher rule, if the `indexName` is set, using the corresponding index to calculate the partition number.
3. return error if the index cannot be found at the runtime, this may happen if the upstream TiDB execute alter index related DDL, the changefeed becomes failed, and should be fixed manually.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
allows user specify the index name for the index-value dispatcher
```

### CODE REVIEW

File: cdc/sink/dmlsink/mq/dispatcher/partition/index_value.go [hunk 1]
```diff
@@ -46,14 +52,30 @@ func (r *IndexValueDispatcher) DispatchRowChangedEvent(row *model.RowChangedEven
...
+			if col.Flag.IsHandleKey() {
+				r.hasher.Write([]byte(col.Name), []byte(model.ColumnValueString(col.Value)))
+			}
 		}
-		if col.Flag.IsHandleKey() {
-			r.hasher.Write([]byte(col.Name), []byte(model.ColumnValueString(col.Value)))
+	} else {
+		names, offsets, ok := row.IndexByName(r.indexName)
+		if !ok {
+			log.Error("index not found",
```
Comment (nongfushanquan) [inline cdc/sink/dmlsink/mq/dispatcher/partition/index_value.go:69]: "index not found when dispatching"

**Review by nongfushanquan — APPROVED** [review]

**Comment by 3AceShowHand** [comment]
/hold

This PR is still under testing, not ready for review yet.

**Comment by CharlesCheung96** [comment]
Please explain the detailed design in pr body.

**Review by nongfushanquan — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**